3627 remove isdev from exposed interface

andrewlee348 opened the pull request.

SamueleA reviewed (changes requested):
There are two improvement that I think would be necessary for this:

1. Can we use have the DEVMODE variable be read from an environment variable?

2. Can we remove the `isDev` parameter from the kitConfig and then replace instances in which isDev would normally be used with that DEVMODE variable?

https://github.com/0xsequence/kit/blob/master/packages/kit/src/types.ts#L78

andrewlee348 commented:
@SamueleA I must've forgotten to push my latest changes. I've removed all isDev from config and kit packages and added a workaround for env variables since the packages aren't able to read env variables from outside the packages folder.

corbanbrook commented on scripts/generate-env.js:
```diff
@@ -0,0 +1,12 @@
+const fs = require('fs')
```
Should not generate an env file that overrides env.ts

corbanbrook replied:
This would potentially get committed into the code base.

corbanbrook commented on packages/kit/src/env.ts:
```diff
@@ -0,0 +1 @@
+export const DEVMODE = false
```
This file should simply reexport the process.env.DEVMODE

corbanbrook replied:
or process.env.DEVMODE should just be used directly instead

corbanbrook commented:
https://turbo.build/repo/docs/crafting-your-repository/using-environment-variables
